File: benchmarks/cross/progs/btree.lua
local function make(d)
  if d == 0 then return {val=1, left=nil, right=nil} end
  return {val=1, left=make(d-1), right=make(d-1)}
end
local function check(n)
  if n.left == nil then return n.val end
  return n.val + check(n.left) + check(n.right)
end
local total = 0
for i=1,8 do total = total + check(make(15)) end
print(total)

File: benchmarks/cross/progs/btree.js
function make(d){ if(d===0) return {val:1,left:null,right:null};
  return {val:1,left:make(d-1),right:make(d-1)}; }
function check(n){ if(n.left===null) return n.val;
  return n.val + check(n.left) + check(n.right); }
let total = 0;
for (let i=0;i<8;i++) total += check(make(15));
console.log(total);
